git-open

```diff
@@ -46,17 +46,37 @@ fi
 # ftp[s]://host.xz[:port]/path/to/repo.git/
 # [user@]host.xz:path/to/repo.git/ - scp-like but is an alternative to ssh.
 
-# Trim "/" and ".git" from the end of the url
-giturl=${giturl%/} giturl=${giturl%.git}
+# Determine whether this is a url (https, ssh, git+ssh...) or an scp-style path
+if [[ "$giturl" =~ ^[a-z\+]+://.* ]]; then
+  # Trim URL scheme and possible username
+  protocol=${giturl%%://*}
+  uri=${giturl#*://}
+  uri=${uri#*@}
+
+  # Split on first '/ to get server name and path
+  domain=${uri%%/*}
+  urlpath=${uri#*/}
+
+  # Remove port number from non-http/https protocols (ie, ssh)
+  if [[ $protocol != "https" && $protocol != "http" ]]; then
+    domain=${domain%:*}
+  fi
+else
+  # Trip possible username from SSH path
+  uri=${giturl##*@}
 
-# Trim before last '@' and protocol (*://) from beginning
-uri=${giturl##*@} uri=${uri##*://}
+  # Split on first ':' to get server name and path
+  domain=${uri%%:*}
+  urlpath=${uri#*:}
+fi
 
-# If there isn't a protocol, we can assume it's using the scp syntax which uses ':' to seperate the path.
-[[ $giturl =~ :// ]] && pathsep='/' || pathsep=':'
+# Trim "/" from beginning of URL; "/" and ".git" from end of URL
+urlpath=${urlpath#/} urlpath=${urlpath%/} urlpath=${urlpath%.git}
 
-# Seperate the domain and the urlpath on the first {pathsep}. This also removes the gitport from the domain.
-domain=${uri%%[:$pathsep]*} urlpath=${uri#*$pathsep}
+# If the URL is provided as "http", preserve that, otherwise assume "https"
+if [[ $protocol != "http" ]]; then
+  protocol="https"
```

Comment:
since we already have https as the default... seems like we shouldn't have to reset it back to that default value.

perhaps in your diff you could set a `gitprotocol` variable .. then here you could just test that and change `protocol` to `"http"` if necessary.

Reply:
Set a `gitprotocol` variable and update `protocol` iff `gitprotocol == 'https'`.